src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -24,47 +24,50 @@ def __init__(self, ivp, gaussfilt, with_smoothing):
 
         Notes
         -----
-        * gaussfilt.dynamicmodel contains the prior,
+        * gaussfilt.dynamic_model contains the prior,
         * gaussfilt.measurementmodel contains the information about the
         ODE right hand side function,
         * gaussfilt.initialdistribution contains the information about
         the initial values.
         """
-        if not issubclass(type(gaussfilt.dynamicmodel), ODEPrior):
-            raise ValueError("Please initialise a Gaussian filter with an ODEPrior")
+        if not issubclass(type(gaussfilt.dynamic_model), pnfs.statespace.Integrator):
+            raise ValueError(
+                "Please initialise a Gaussian filter with an Integrator (see filtsmooth.statespace)"
+            )
         self.gfilt = gaussfilt
         self.sigma_squared_mle = 1.0
         self.with_smoothing = with_smoothing
         super().__init__(ivp)
 
     def initialise(self):
-        return self.ivp.t0, self.gfilt.initialrandomvariable
+        return self.ivp.t0, self.gfilt.initrv
 
-    def step(self, t, t_new, current_rv, **kwargs):
+    def step(self, t, t_new, current_rv):
         """Gaussian IVP filter step as nonlinear Kalman filtering with zero data."""
-        # 0. Obtain the diffusion matrix; required for calibration / error estimation
-        discrete_dynamics = self.gfilt.dynamod.discretise(t_new - t)
-        diffmat = discrete_dynamics.diffusionmatrix(t_new)
+
+        # Read the diffusion matrix; required for calibration / error estimation
+        discrete_dynamics = self.gfilt.dynamic_model.discretise(t_new - t)
+        diffmat = discrete_dynamics.diffmat
 
         # 1. Predict
-        pred_rv, _ = self.gfilt.predict(t, t_new, current_rv, **kwargs)
+        pred_rv, _ = self.gfilt.predict(t, t_new, current_rv)
```

Comment:
the kwargs are resolved in `filtsmooth` and consequently, not needed here anymore.